src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -298,6 +298,7 @@ bool AccountStoreSC<MAP>::UpdateAccounts(const uint64_t& blockNum,
       // generate address for new contract account
       Address contractAddress =
           Account::GetAddressForContract(fromAddr, fromAccount->GetNonce());
+      toAddr = contractAddress;
```

Comment:
Thanks, well spotted.

Could we resolve this by renaming the other variable down below to `contractAddress`.  toAddr should mean the address to which the transaction was sent, which would be zero in this case.